algebra/tests/matrix/various.c

```diff
@@ -483,3 +483,136 @@ TEST_GROUP_RUNNER(group_matrix_times)
 	RUN_TEST_CASE(group_matrix_times, matrix_times_nan);
 	RUN_TEST_CASE(group_matrix_times, matrix_times_nanTrp);
 }
+
+
+/* ##############################################################################
+ * -----------------        matrix_writeSubmatrix tests       -------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_matrix_writeSubmatrix);
+
+
+TEST_SETUP(group_matrix_writeSubmatrix)
+{
+	/* M1 = F */
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&M1, buffs_rowsF, buffs_colsF, buffs_F, buffs_colsF * buffs_rowsF));
+
+	/* M2 = B */
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&M2, buffs_rowsB, buffs_colsB, buffs_B, buffs_colsB * buffs_rowsB));
+
+	/* M3 = B */
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, algebraTests_matrixCopy(&M3, &M1));
+}
+
+
+TEST_TEAR_DOWN(group_matrix_writeSubmatrix)
+{
+	matrix_bufFree(&M1);
+	matrix_bufFree(&M2);
+	matrix_bufFree(&M3);
+}
+
+
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_onStart)
+{
+	int rowToWrite = 0, colToWrite = 0;
+
+	TEST_ASSERT_EQUAL_INT(WRITE_SUBMAT_OK, matrix_writeSubmatrix(&M1, rowToWrite, colToWrite, &M2));
+
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTests_writeSubmatCheck(&M3, rowToWrite, colToWrite, &M2, &M1));
+}
+
+
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_inMiddle)
+{
+	int rowToWrite = 5, colToWrite = 2;
+
+	TEST_ASSERT_EQUAL_INT(WRITE_SUBMAT_OK, matrix_writeSubmatrix(&M1, rowToWrite, colToWrite, &M2));
+
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTests_writeSubmatCheck(&M3, rowToWrite, colToWrite, &M2, &M1));
+}
+
+
+/* This test checks if function changes `Src` matrix */
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_sourceRetain)
+{
+	matrix_bufFree(&M3);
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, algebraTests_matrixCopy(&M3, &M2));
+
+	TEST_ASSERT_EQUAL_INT(WRITE_SUBMAT_OK, matrix_writeSubmatrix(&M1, 0, 0, &M2));
+
+	/* M2 and M3 should be the same */
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTest_equalMatrix(&M3, &M2));
+}
+
+
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_fullWrite)
+{
+	matrix_bufFree(&M2);
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, algebraTests_matrixCopy(&M2, &M1));
+
+	TEST_ASSERT_EQUAL_INT(WRITE_SUBMAT_OK, matrix_writeSubmatrix(&M1, 0, 0, &M2));
+
+	/* Matrix A and B have the same sizes, so after witeSubmatrix M1 should be equal to M2 */
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTest_equalMatrix(&M1, &M2));
+}
+
+
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_tooFewCols)
+{
+	int rowToWrite = 0, colToWrite = M1.cols - M2.cols / 2;
```

Comment:
Split for better readibility
```suggestion
	unsigned int row, col; 

	row = 0;
	col = M1.cols - M2.cols / 2;
```